File: json-to-messages/ruby/features/parameters/parameters_steps.rb
Given('I have {int} cucumbers') do |count|
  @cucumbers = count
end

When('I eat {int} cucumbers') do |eaten|
  @cucumbers -= eaten
end

Then('I have {int} cucumbers left') do |left|
  expect(@cucumbers).to eq(left)
end

Given('a step with a doctring:') do |string|
end

When('a step with a datatable') do |table|
end

File: json-to-messages/javascript-testdata/features/parameters/parameters_steps.ts
import assert from 'assert'
import { Given, When, Then, setWorldConstructor } from '@cucumber/cucumber'

function CustomWorld() {
  this.cucumbers = 0
}

setWorldConstructor(CustomWorld)

Given('I have {int} cucumbers', function (count: number) {
  this.cucumbers = count
})

When('I eat {int} cucumbers', function (eaten: number) {
  this.cucumbers -= eaten
})

Then('I have {int} cucumbers left', function (left: number) {
  assert.strictEqual(this.cucumbers, left)
})

Given('a step with a doctring:', function () {
  // no-op
})

When('a step with a datatable', function () {
  // no-op
})
